contracts/governance/Vesting/VestingLogic.sol

```diff
@@ -110,26 +110,35 @@ contract VestingLogic is IVesting, VestingStorage, ApprovalReceiver {
     }
 
     /**
-     * @notice Withdraws all tokens from the staking contract and
+     * @notice Withdraws unlocked tokens from the staking contract and
      * forwards them to an address specified by the token owner.
      * @param receiver The receiving address.
-     * @dev Can be called only by owner.
-     * @dev **WARNING** This function should not be no longer used by Sovryn Protocol.
-     * Sovryn protocol will use the cancelTeamVesting function for the withdrawal moving forward.
      * */
-    function governanceWithdrawTokens(address receiver) public {
-        require(msg.sender == address(staking), "unauthorized");
-
-        _withdrawTokens(receiver, true);
+    function withdrawTokens(address receiver) public onlyOwners {
+        uint256 startFrom = startDate + cliff;
+        _withdrawTokens(receiver, false, startFrom, block.timestamp);
     }
 
     /**
-     * @notice Withdraws unlocked tokens from the staking contract and
+     * @notice Withdraws unlocked tokens partially (based on the max withdraw iteration that has been set) from the staking contract and
      * forwards them to an address specified by the token owner.
      * @param receiver The receiving address.
+     * @param startFrom The start value for the iterations.
+     * @param maxWithdrawIterations max withdrawal iteration
      * */
-    function withdrawTokens(address receiver) public onlyOwners {
-        _withdrawTokens(receiver, false);
+    function withdrawTokensStartingFrom(
+        address receiver,
+        uint256 startFrom,
+        uint256 maxWithdrawIterations
+    ) public onlyOwners {
+        uint256 defaultStartFrom = startDate + cliff;
+
+        startFrom = startFrom < defaultStartFrom ? defaultStartFrom : startFrom;
```

Comment:
sorry, that must have been confusing   
lm clarify  
the original `_withdrawTokens` function used this cycle:  
`for (uint256 i = startDate + cliff; i <= end; i += FOUR_WEEKS) {`   
meaning that to keep consistent, startForm parameter in the updated function should be adjusted to `startDate + cliff + n * FOUR_WEEKS` - you can take this adjustment logic from the `timestampToLockDate` in Staking.   
if it is not, further calculations would be messed up.  
i also suggest to move all the code and add this adjustment to the correct date, from this function to `_withdrawTokens` to avoid potential misuse of `startFrom` parameter and add fuzzy-like tests passing a few random startFrom parameters as well as corner cases and compare with the expected result.

Reply:
updated in this [commit](https://github.com/DistributedCollective/Sovryn-smart-contracts/pull/529/commits/63301217172621a0f99fa34d4787b5d25c42444e)